frontend/packages/operator-lifecycle-manager/src/components/create-operand.tsx

```diff
@@ -639,6 +649,15 @@ export const CreateOperandForm: React.FC<CreateOperandFormProps> = (props) => {
     new Set<SpecCapability.fieldGroup>(),
   );
 
+  //if 'urn:alm:descriptor:com.tectonic.ui:fieldGroup:example:expand' and 'urn:alm:descriptor:com.tectonic.ui:fieldGroup:example' exist, remove 'urn:alm:descriptor:com.tectonic.ui:fieldGroup:example'.
```

Comment:
Seems better to use a separate `x-descriptor` (capability) to indicate whether the field group should be expanded by default.

Reply:
@alecmerdler Thanks for your suggestion, I have thought about using a separate x-descriptor to indicate whether the field group should be expanded by default, but according to the related docs, one x-descriptor is used to define one component, it also define the status the component. If we use 2 x-descriptors to define one component, does this break the rule? 

Or maybe I misunderstood something. If you could provide an example the x-descriptor you mentioned above(the one to indicate whether the field group should be expanded by default), I will be highly appreciated! Thanks.

Reply:
I guess our docs are misleading; usually, you just need 1 `x-descriptor` to create a component. But you should think of them more as _capabilities_ (that's what I think in my head when I read `x-descriptor`). Each field has a list of _capabilities_, and it's up to the UI to figure out which ones it supports. One example is `urn:alm:descriptor:com.tectonic.ui:select:`.

It's easier for Operator authors to use `urn:alm:descriptor:com.tectonic.ui:advanced` to collapse `fieldGroups` and leave them expanded by default without another `x-descriptor`.

Reply:
> I guess our docs are misleading; usually, you just need 1 `x-descriptor` to create a component. But you should think of them more as _capabilities_ (that's what I think in my head when I read `x-descriptor`). Each field has a list of _capabilities_, and it's up to the UI to figure out which ones it supports. One example is `urn:alm:descriptor:com.tectonic.ui:select:`.

Thanks for clarification. If we consider x-descriptor as _capabilities_ then we may want to have a stand alone _fieldGroup_ component, since the x-descriptors should define what the _field_ can do or the status of it. For example, 'urn:alm:descriptor:com.tectonic.ui:text' can define a _Input_ component; 'urn:alm:descriptor:com.tectonic.ui:fieldGroup:' can define a fieldGroup component. If we want the _fieldGroup_ to be expanded, we should add a x-descriptor like 'urn:alm:descriptor:com.tectonic.ui:expand' to the _fieldGroup_ _capabilities_ instead of the _field capabilities_, right? And in the PR, I have already made the _fieldGroup_ a stand alone component.



> It's easier for Operator authors to use `urn:alm:descriptor:com.tectonic.ui:advanced` to collapse `fieldGroups`
I am afraid this may lead to the argument about whether _advanced_ group should be collapsed by default, some users may say yes, but some users may say no. So, why don't we give them the ability to explicitly configure the expanded/collapsed status? For example, 'urn:alm:descriptor:com.tectonic.ui:fieldGroup:groupName:expand' makes the group expanded, or ''urn:alm:descriptor:com.tectonic.ui:fieldGroup:groupName'' and 'urn:alm:descriptor:com.tectonic.ui:expand' together make the group expanded.



> leave them expanded by default without another `x-descriptor`.
 I am not sure whether leaving the groups expanded by default is the best option, since from UI/UX aspect, Operators with many fields will have a long list in both of the Create Operand Form and the Operand Form.
Here is an example, [https://console-openshift-console.apps.mmikhail-mwocp42.devcluster.openshift.com/k8s/ns/swati/clusterserviceversions/3scale-operator.v0.3.0/capabilities.3scale.net~v1alpha1~API/~new](https://console-openshift-console.apps.mmikhail-mwocp42.devcluster.openshift.com/k8s/ns/swati/clusterserviceversions/3scale-operator.v0.3.0/capabilities.3scale.net~v1alpha1~API/~new)

Your feedback will be highly appreciated.